src/Distribution/Nixpkgs/PackageMap.hs

```diff
@@ -21,11 +22,14 @@ import Control.Lens
 type PackageMap = Map Identifier (Set Path)
 
 readNixpkgPackageMap :: FilePath -> Maybe Path -> IO PackageMap
-readNixpkgPackageMap nixpkgs attrpath = fmap identifierSet2PackageMap (readNixpkgSet nixpkgs attrpath)
```

Comment:
I believe now that it would be better to open up that function's API to the notion of passing arbitrary arguments to `nix-env`. b87bfe0d26e0dd9b219c2919b9ccd24993371241 is a step into that direction and I'll run some experiments to see how that feels in practice.